File: stem-explorer-ng/src/app/shared/models/challenge-level.ts
import { AnswerType } from '../enums/answer-type.enum';

export interface ChallengeLevel {
    uid: number;
    questionText: string;
    difficulty: number;
    instructions: string;
    answerType: AnswerType;
    possibleAnswers: string[];
    answer: string[];
    challengeId: number;
    hint: string;
}

File: StemExplorerAPI/StemExplorerAPI/Models/Entities/ChallengeLevel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StemExplorerAPI.Models.Entities
{
    public class ChallengeLevel
    {
        public int Id { get; set; }
        public string QuestionText { get; set; }
        public Enums.ChallengeDifficulty Difficulty { get; set; }
        public string Instructions { get; set; }
        public Enums.AnswerType AnswerType { get; set; }
        public string Hint { get; set; }
        public string VideoEmbedUrl { get; set; }

        public List<string> PossibleAnswers { get; set; }
        public List<string> Answers { get; set; }

        // EF relationship definition
        public int ChallengeId { get; set; }
        public Challenge Challenge { get; set; }
    }
}
